evap/staff/templates/staff_course_comments_quick.html

```diff
@@ -0,0 +1,403 @@
+            slider.on("touchcancel", resetTouchSliding);
+
+            function resetTouchSliding() {
+                touchX = null;
+                findActive(2).css({transform: "", opacity: ""});
+            }
+
+            handle("repeat-start");
+        });
+
+        function handle(action, click) {
+            let parts = action.split("-");
+
+            if(parts[0] === "slide") {
+                slide(parts[1]);
+                updateButtons();
+            } else if(parts[0] === "repeat") {
+                repeat(parts[1]);
+                updateButtons();
+            } else if(action === "next-course" && !click) {
+                slider.find("[data-action=next-course]")[0].click();
+            } else if(["publish", "make_private", "hide", "unreview"].indexOf(action) !== -1) {
+                press(action, click);
+            }
+
+            updateState();
+        }
+
+        function slide(direction) {
+            if(direction === "left" && index > 0) {
+                index--;
+            } else if(direction === "right" && index < items.length) {
+                index++;
+            }
+
+            if(index < items.length) {
+                active = items[index];
+                slideLayer(2, direction, active);
+            } else {
+                active = null;
+                slideLayer(2, direction, slider.find(".alert"))
+            }
+        }
+
+        function slideLayer(layer, direction, element) {
+            // to preserve the vertical positions and heights during transition,
+            // the elements will be inactivated from bottom to top and activated from top to bottom
+
+            if(element && element.is(".active")) {
+                return;
+            }
+
+            let last = findActive(layer);
+            if(last.length !== 0) {
+                let reversed = direction === "left"? "right" : "left";
+                last.css({top: last.position().top, height: last.outerHeight()});
+                last.removeClass("active to-left to-right");
+                last.addClass("to-" + reversed);
+            }
+
+            if(layer > 0) {
+                if(active) {
+                    let reference = element.prevAll("[data-layer=" + (layer - 1) + "]").first();
+                    slideLayer(layer - 1, direction, reference);
+                } else {
+                    slideLayer(layer - 1, direction, null);
+                }
+            }
+
+            if(element !== null) {
+                element.removeClass("to-left to-right");
+                element.addClass("to-" + direction);
+                element.position();
+                element.addClass("active");
+            }
+        }
+
+        function findActive(layer) {
+            if(layer === 2) {
+                return slider.find("[data-layer=2].active, .alert.active");
+            } else {
+                return slider.find("[data-layer=" + layer + "].active");
+            }
```

Comment:
Oh my… You found my greatest struggle point. Semantically, an `.slider-item[data-layer]` is something with real content, so `0` is questionnaires, `1` is questions and `2` is answers. So what are these alerts? They are like a substitute to all three layers. So they need to be slid away before a layer 2 is slid in (otherwise they will be pushed down), but they are no `[data-layer=2]` (read: no answer). `findActive()` is some poor naming but I'm still waiting for a nice idea